official/utils/logging/benchmark_uploader.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Library to upload benchmark generated by BenchmarkLogger to remote repo.
+
+This library require google cloud bigquery lib as dependency, which can be
+installed with:
+  > pip install --upgrade google-cloud-bigquery
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import sys
+import uuid
+
+from google.cloud import bigquery
+
+import tensorflow as tf # pylint: disable=g-bad-import-order
+
+from official.utils.logging import logger
+
+
+FLAGS = None
+
+
+class BigQueryUploader(object):
```

Comment:
I have a conceptual proposal:
  1) Pull dataset_name and run_id into `__init__()`. In effect make all local information part of the class state.
  2) Support intermediate uploading. So if I call upload_metric() with 500 lines, and then call it again with 1000 the second call will upload lines 501 to 1000.

The reason is that is would allow intermediate uploading during the training loop, rather than at the very end. Depending on the sort of monitoring we want to build on top of BigTable that could be desirable. Discuss.